cockpit/src/subscriptions-client.js

```diff
@@ -428,28 +469,40 @@ client.getSubscriptionStatus = function(dfd = null) {
     gettingStatus = true;
 
     entitlementService.wait(() => {
-        entitlementService.GetStatus('', userLang)
-            .then(result => {
-                const status = JSON.parse(result);
-                client.subscriptionStatus.status = status.status;
-                if (this.dfd && client.closeRegisterDialog) {
-                    this.dfd.resolve();
-                    client.closeRegisterDialog = false;
-                }
-            })
-            .catch(() => {
-                client.subscriptionStatus.status = 'Unknown';
-            })
-            .then(() => {
-                gettingStatus = false;
-                getSubscriptionDetails();
-                needRender();
-            });
+        safeDBusCall(entitlementService, 'GetStatus', function () {
```

Comment:
I would use arrow function here too `() => {};`. It will not be necessary to do explicit binding of `this`.  You will be able to use `this` instead of `that` in following code.